File: src/main/java/commands/cooldownchecker/CooldownManager.java
package commands.cooldownchecker;

import com.google.common.cache.CacheBuilder;
import com.google.common.cache.CacheLoader;
import com.google.common.cache.LoadingCache;
import constants.Settings;
import org.checkerframework.checker.nullness.qual.NonNull;

import java.util.concurrent.TimeUnit;

public class CooldownManager {

    private static final CooldownManager ourInstance = new CooldownManager();

    public static CooldownManager getInstance() {
        return ourInstance;
    }

    private CooldownManager() {
    }

    private final LoadingCache<Long, CooldownUserData> cooldownUserDataMap = CacheBuilder.newBuilder()
            .expireAfterAccess(Settings.COOLDOWN_TIME_SEC, TimeUnit.SECONDS)
            .build(new CacheLoader<>() {
                @Override
                public CooldownUserData load(@NonNull Long userId) {
                    return new CooldownUserData();
                }
            });

    public synchronized CooldownUserData getCooldownData(long userId) {
        return cooldownUserDataMap.getUnchecked(userId);
    }

}

File: src/main/java/commands/cooldownchecker/CoolDownManager.kt
package commands.cooldownchecker

import com.google.common.cache.CacheBuilder
import com.google.common.cache.CacheLoader
import constants.Settings
import java.util.concurrent.TimeUnit

object CoolDownManager {

    private val coolDownUserDataMap = CacheBuilder.newBuilder()
        .expireAfterAccess(Settings.COOLDOWN_TIME_SEC.toLong(), TimeUnit.SECONDS)
        .build(object : CacheLoader<Long, CoolDownUserData>() {
            override fun load(userId: Long): CoolDownUserData {
                return CoolDownUserData()
            }
        })

    @JvmStatic
    @Synchronized
    fun getCoolDownData(userId: Long): CoolDownUserData {
        return coolDownUserDataMap.getUnchecked(userId)
    }

}